### PR TITLE
fix ci error

### PR DESCRIPTION
**Pull request by martgras**
fix ci error (make compare operator const function)
update comments
improving handling expired commands (not directly related to your changes)

## Description:
Change suggestions

### CODE REVIEW

File: esphome/components/modbus_controller/modbus_controller.h [hunk 1]
```diff
@@ -281,8 +280,12 @@ class SensorItemsComparator {
       return lhs->start_address < rhs->start_address;
     }
 
-    // The pointer to the sensor are added at last to ensure that
-    // multiple sensors with the save values can be added with a stable sort order.
+    // sort by bitmask address
```
Comment (stegm) [inline esphome/components/modbus_controller/modbus_controller.h]: Can you explain, why sorting by bitmask is important? For my understanding it does not play a role. Please note the last comparison on the return statement ensures that two sensors with the same address, function etc. can be added to the set.

File: esphome/components/modbus_controller/modbus_controller.cpp [hunk 2]
```diff
@@ -24,15 +24,22 @@ bool ModbusController::send_next_command_() {
...
 
-    ESP_LOGV(TAG, "Sending next modbus command to device %d register 0x%02X count %d", this->address_,
-             command->register_address, command->register_count);
-    command->send();
-    this->last_command_timestamp_ = millis();
-    // remove from queue if no handler is defined or command was sent too often
-    if (!command->on_data_func || command->send_countdown < 1) {
-      ESP_LOGD(TAG, "Modbus command to device=%d register=0x%02X countdown=%d removed from queue after send",
-               this->address_, command->register_address, command->send_countdown);
+    // remove from queue if command was sent too often
```
Comment (stegm) [inline esphome/components/modbus_controller/modbus_controller.cpp:27]: Basically I'm fine with this change, but in my experience small, focused PR are preferred by the maintainers.

File: esphome/components/modbus_controller/modbus_controller.cpp [hunk 3]
```diff
@@ -277,12 +280,13 @@ void ModbusController::dump_config() {
 #if ESPHOME_LOG_LEVEL >= ESPHOME_LOG_LEVEL_VERBOSE
   ESP_LOGCONFIG(TAG, "sensormap");
   for (auto &it : sensorset_) {
-    ESP_LOGCONFIG(TAG, "  Sensor start=0x%X count=%d size=%d", it->start_address, it->register_count,
+    ESP_LOGCONFIG(TAG, " Type=%zu Sensor start address=0x%X, offset=0x%X count=%d size=%d",
```
Comment (stegm) [inline esphome/components/modbus_controller/modbus_controller.cpp]: Not sure, but I would prefer `Sensor type=%zu start address=0x%x...` just to be similar to the log message of the range at line 289.

**Review by stegm — CHANGES_REQUESTED** [review]
Great, just some comments/suggestions.

File: esphome/components/modbus_controller/modbus_controller.h (hunk 1 above)
Comment (martgras) [reply esphome/components/modbus_controller/modbus_controller.h]: I had scenario where it did matter but to be honest can't remember. Probably not required anymore 
Removing it..

**Review by stegm — APPROVED** [review]

**Pull request merged**